### PR TITLE
style: add template for spike issue

### PR DESCRIPTION
**Pull request by HectorMRC**
## Intent

This PR aims to embrace all the spike issues under the same template.

## Checklist

- [x] PR is limited to a single logical change.
- [x] At least the first commit has a clear title and a descriptive message containing reasoning for the change. This will be included in the final commit message after squashing and merging.

### CODE REVIEW

**Comment by alexveecle** [comment]
I think the PR template cannot have "conclusions/observations", because when the PR is opened, we have not done the spike yet.

I would add instead a timeframe for the spike, so it's limited in time. Perhaps a section to include things which are explicitly not in scope.

**Comment by HectorMRC** [comment]
I absolutely agree with the timeframe section, I was thinking about it but wasn't sure. I also like the "out of scope" section.

About the "conclusions/observations" section: then I misunderstood how spikes work. From my perspective the issue is meant to be created once the spike is finished, being the issue a kind of "report" of the spike process and conclusions. There is no PR at all, only the issue. I am wrong @vE5li ?

**Comment by alexveecle** [comment]
Ah, in that case it would make sense. However, I think we should put something in the sprints to represent the spikes. But it could be a draft card without an issue.

**Comment by HectorMRC** [comment]
That's exactly what I have done jeje :)
There are draft cards in the project board representing every spike

**Review by alexveecle — APPROVED** [review]

File: .github/SPIKE_ISSUE_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+# Timeframe
+<!---
+How long (in hours) did the spike take?
+-->
+
+# Development
+<!---
+Which has been the development/testing process?
+-->
```
Comment (vE5li) [inline .github/SPIKE_ISSUE_TEMPLATE.md:24]: Can you maybe elaborate what this section is for?

**Comment by vE5li** [comment]
> About the "conclusions/observations" section: then I misunderstood how spikes work. From my perspective the issue is meant to be created once the spike is finished, being the issue a kind of "report" of the spike process and conclusions. There is no PR at all, only the issue. I am wrong @vE5li ?

It's not really part of the definition of a spike when to create issues (or really whether to use issues at all). I personally don't mind creating the issue after the spike is over, but I would prefer having some sections of the template pre-filled with _`will be updated once the spike is over`_. So basically we start spikes as issues with your template, fill in everything that is known ahead of time, and update the issue once the spike is over.

**Review by vE5li — APPROVED** [review]

**Pull request merged**